### DISCUSSION TITLE
Batch file processing

### DISCUSSION THREAD

**Discussion by lenover12** [voicepaw/so-vits-svc-fork#521] [Q&A] [answered]
I have trained a model on my voice and I want to batch convert thousands of small audio files to make a game mod. Is there a way to do this?

assuming I need to use the CLI, can I choose a pre-set I created?

**Comment by lenover12** [comment]
I found a solution with the help and annoyance of chatGPT
my goal was to find every .ogg file in a parent and its child directories and process them with the voice model, outputting them with their directory path in so-vits-svc-fork\infer_output\ + the directory path

Create a .bat file like this for use on windows 10: (change the paths as you want, and the arguments)
`cd D:\so-vits-svc-fork
SETLOCAL ENABLEDELAYEDEXPANSION
FOR /R "C:\Users\user\Desktop\parent_directory" %%f IN (*.ogg) DO (
    set "rel_path=%%~pf"
    set "rel_path=!rel_path:~14,-1!"
    set "out_path=D:\so-vits-svc-fork\infer_output\!rel_path!\%%~nf.ogg"
    MD "!out_path:\%%~nxf=!" 2>NUL
    svc infer "%%f" --output-path "!out_path!" --speaker "lenover" --f0-method "crepe" --transpose -15 --pad-seconds 0.1
)`

then in your virtual environment in command prompt (as administrator) run the .bat file

**Reply by nikich340** [reply]
That's not a real solution, inferencing every files separately requires model to be loaded/unloaded to gpu every time.

**Reply by lenover12** [reply]
before batch processing was added it helped me convert thousands of files, so If anyone else needed a quick and dirty solution I shared what I got working.

**Comment by GarrettConway** [comment]
Wow, I'm amazed that worked out for you. I'll take a first pass at adding in a simple CLI batch convert tonight so hopefully this isn't as painful in the future.

**Comment by nikich340** [comment, marked as answer]
Currently GUI supports selecting a folder for inference, what equals to batch processing.
